posts/hybrid-cloud-cloud-responsibly.md

```diff
@@ -0,0 +1,94 @@
+
+The microservices are load balanced by Fabio, a fine grained application router that routes subsets of URLs to different instances of micro services. The service discovery engine, Consul, finds services in the network, health checks them, and notifies Fabio of what services are healthy/available. The container technology, Nomad, defines services to be deployed (Docker) for Consul assigning them names, ports, and tags to tell Fabio what type of service it is.
+
+The arrowed lines represent the request flow through the deployed application. The user interface is a single page application (SPA) that obtains content from the gateway and submits REST calls for data to the gateway. The gateway forwards requests to Fabio which are then load balanced and routed to the appropriate microservice to be serviced. Additionally, microservices can leverage Kafka streams to support queuing and asynchronous processing for applications having those needs.
+
+<h2>Example</h2>
+Hybrid Cloud is about running your infrastructure where it makes sense. For example, the diagram below shows an application in AWS so that it’s fault tolerant, and it uses Fabio which routes to microservices either on-prem or in the cloud depending on whether it’s a compute-heavy request (cloud) or something simple like a lookup (on-prem). While the federated directory and user authentication is kept private on-prem.
+
+
+![alternate text](https://github.com/ippontech/blog-usa/blob/master/images/2019/08/IpponWay-Hybrid-Cloud-Example.png)
+
+
+<h2>Software Selection</h2>
+Ippon selected the following software for our solution:
+
+- Scheduler/Orchestration - Nomad
+- Service Discovery - Consul
+- Configuration/Secrets - Vault, Consul
+- Health/Performance Monitoring - Prometheus
+- App Router/Load Balancer - Fabio
+- Role-based Access Control (Authorization) - Spring OAuth - Keycloak
+- Authentication - Keycloak OpenId Connect
+- Code generation - JHipster
+- Data modeling - JHipster JDL
+- Microservice Framework - Spring Boot
+- Front-end development framework/language - Angular/JavaScript/TypeScript
+- Back-end/Microservice framework/language - Spring Boot/Java
+- User Account Management - Keycloak
+- Cloud Deployment - Nomad, Ansible, Docker
+- Queuing/Streaming - Kafka
+
+A key aspect of the above software selection was the preference for open source software over commercial software. For example, this guided the choice of Keycloak over Okta for
+authentication and authorization services. HashiCorp software (Nomad, Consul) was selected to obtain the abstraction layer to insulate the solution from the specifics of individual cloud providers and on-prem. JHipster was selected because of our company’s expertise in using it to quickly produce enterprise grade applications. The remaining software selections were guided by popularity in the industry and proven reliability.
```

Comment:
I would like to re-word the reason for using JHipster. I would rather say because it is a great open source tool that generates enterprise grade applications. (or similar) I like mentioning that we are experts but I think it sounds better if we are making that choice based on the merits of the tool and not because we are experts.